### PR TITLE
feat: replacing windicss by tailwindcss in hoppscotch-ui

### PR DESCRIPTION
**Pull request by anwarulislam**
Closes HFE-100 HFE-101 HFE-97 HFE-259 HFE-265

## Description:

The PR replaces WindiCSS with TailwindCSS from all hoppscotch frontend projects. 

## Things changed:

- Removed `windicss` package from `hoppscotch-ui`, `hoppscotch-common`, `hoppscotch-selfhost-web`
- Removed `windicss` configs
- Added a new `config` package to the packages directory. The directory contains common config files. Such as `prettier`, `eslint`, `tailwind`, `postcss`.
- Added `TailwindCSS` to all frontend packages.
- Fixes WindiCSS related issue existed after installing Tailwind

### CODE REVIEW

File: packages/hoppscotch-ui/src/plugin.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import type { Plugin, App } from "vue"
 
 import "./assets/scss/styles.scss"
```
Comment (liyasthomas) [inline packages/hoppscotch-ui/src/plugin.ts:3]: Import the newly created tailwind directives file here.
```
import "./assets/scss/main.scss"
```

**Review by liyasthomas — CHANGES_REQUESTED** [review]

**Review by liyasthomas — CHANGES_REQUESTED** [review]
In `hoppscotch-ui/histoire.setup.ts`;

Should we replace
```
import "virtual:windi.css"
```
with the newly created tailwind directives file:
```
import "./src/assets/scss/main.scss"
```

**Comment by liyasthomas** [comment]
I'm converting this PR to a draft since I could find a few WindiCSS classes that are not being ported to Tailwind-flavoured styles. Moreover, the Histoire module seems to be broken. eg: `pnpm run story:dev` from `packages/hoppscotch-ui`.

Here's a screenshot of the Modal component:

Screenshot 2023-05-24 at 7 54 01 AM

**Comment by AndrewBastin** [comment]
Make sure this PR has a backing Linear ticket guys! @anwarulislam

**Review by nivedin — CHANGES_REQUESTED** [review]
there are some windi/tw colors to be migrated, like [here](https://github.com/hoppscotch/hoppscotch/blob/41232626ae7f546fb26920107c9064acfbfc7a7a/packages/hoppscotch-ui/src/assets/scss/styles.scss#L346)

**Comment by anwarulislam** [comment]
@AndrewBastin I have added a new package called `config` to keep our config-related stuff there. such is prettier, eslint, tailwind, postcss.

Do you have any suggestions regarding this?

**Comment by liyasthomas** [comment]
> @AndrewBastin I have added a new package called `config` to keep our config-related stuff there. Such as prettier, eslint, tailwind, postcss.
> 
> Do you have any suggestions regarding this?

cc: @AndrewBastin

File: packages/hoppscotch-ui/src/assets/themes/font-sizes.scss [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  --upper-mobile-secondary-sticky-fold: 8.688rem;
+  --upper-mobile-sticky-fold: 10.75rem;
+  --upper-mobile-tertiary-sticky-fold: 8.25rem;
+  --lower-primary-sticky-fold: 3rem;
+  --lower-secondary-sticky-fold: 5.063rem;
+  --lower-tertiary-sticky-fold: 7.125rem;
+  --sidebar-primary-sticky-fold: 2rem;
+}
+
+@mixin font-medium {
```
Comment (JoelJacobStephen) [inline packages/hoppscotch-ui/src/assets/themes/font-sizes.scss:17]: We have removed support for different font sizes in the app. Hence this file can be removed.

File: packages/hoppscotch-ui/src/assets/themes/themes.scss [hunk 3]
```diff
@@ -0,0 +1,77 @@
+@import "./base-themes.scss";
+@import "./editor-themes.scss";
+@import "./accent-themes.scss";
+@import "./font-sizes.scss";
```
Comment (JoelJacobStephen) [inline packages/hoppscotch-ui/src/assets/themes/themes.scss:4]: We can remove this import as font sizes aren't supported

File: packages/hoppscotch-ui/src/assets/themes/themes.scss [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+:root[data-font-size="small"] {
+  @include font-small;
+}
+
+:root[data-font-size="medium"] {
+  @include font-medium;
+}
+
+:root[data-font-size="large"] {
```
Comment (JoelJacobStephen) [inline packages/hoppscotch-ui/src/assets/themes/themes.scss:75]: We don't need to support different font sizes

File: packages/hoppscotch-ui/src/components/smart/Modal.vue [hunk 5]
```diff
@@ -17,21 +17,21 @@
...
+              class="absolute inset-0 bg-primaryLight opacity-80 focus:outline-none"
               tabindex="0"
               @click="!dialog ? close() : null"
             ></div>
           </div>
         </Transition>
         <span
           v-if="placement === 'center'"
-          class="sm:h-screen <sm:hidden sm:align-middle"
+          class="sm:h-screen sm:align-middle <sm:hidden"
```
Comment (JoelJacobStephen) [inline packages/hoppscotch-ui/src/components/smart/Modal.vue:28]: Is this a typo?

**Review by JoelJacobStephen — CHANGES_REQUESTED** [review]
The UI in the different parts of the Admin dashboard have few issues such as an enlarged logo in the login page, compressed sidebar, mobile navigation toggle being visible in normal screen size, profile icon in single individual user page being dragged to full width and so on. These UI inconsistencies have to be addressed. Support for font sizes also have to be dropped.

**Comment by JoelJacobStephen** [comment]
On the app side, there are UI inconsistencies such as the tabs not having colors, spotlight search width being extended, buttons such as invite people to workspace and workspace selector not having color fill, request selector having extended width, indentation when creating a collection and a folder inside sticks to the left border and so on....

**Review by nivedin — CHANGES_REQUESTED** [review]
image

realtime > mqtt > New Subscription

the colour picking button needs alignment fix

**Comment by nivedin** [comment]
@anwarulislam can you check if decimal values used are woking properly because as @JoelJacobStephen mentioned the tree structure indentation uses decimal values, also there are some places where this is happening.

File: packages/hoppscotch-ui/src/components/smart/Modal.vue (hunk 5 above)
Comment (anwarulislam) [reply packages/hoppscotch-ui/src/components/smart/Modal.vue:28]: No, this is not a typo.

**Comment by anwarulislam** [comment]
> also can you check if [this ](https://github.com/hoppscotch/hoppscotch/pull/3076#pullrequestreview-1477330680)is resolved

It's resolved

**Review by JoelJacobStephen — APPROVED** [review]
LGTM from the Admin Dashboard side!!

**Review by nivedin — APPROVED** [review]
LGTM 💯

**Review by jamesgeorge007 — CHANGES_REQUESTED** [review]
- The send request button in the `REST` page appears to have become smaller in width.

  #### Before
  send-button-before

  #### After
  send-button-after

- The right pane within Authorization, Pre-request Script, etc, where we link to the documentation, has grown in width.

  #### Before  
  right-pane-before

  #### After
  right-pane-after

- The `Create new team` button in the workspace selector has become slightly larger.

  #### Before
  workspace-selector-before
  
  #### After
  workspace-selector-after

- The team cards rendered under the profile page have grown in width.

  ### Before
  profile-teams-before

  ### After
  profile-teams-after

- On mobile, there's an extra scrollable area for certain views, and the editor part for Pre-request Script and Tests appears collapsed. Also, the text doesn't show up (only icons are rendered) for the nav items below (REST, GraphQL, Realtime, and Settings).

  > Pixel 5 (393 x 851)

  https://github.com/hoppscotch/hoppscotch/assets/25279263/b44d4ac4-2a38-45a0-b5ca-0a838e570dc4

If possible can we also fix the issue about the navigation elements getting rendered twice on iPad resolution (768 x 1024)?

image

File: packages/hoppscotch-backend/mailcathcer.service
Comment (AndrewBastin) [inline packages/hoppscotch-backend/mailcathcer.service:1]: ?

**Review by jamesgeorge007 — COMMENTED** [review]
- Minor nitpick: the difference in height (48 - 50px) for the header persists.
  
  #### Before

  image

  #### After

  image

- Duplicate rendering of nav items in iPad resolution; can be revisited separately since it wasn't introduced in this PR.

  image

- Spotted a difference in sizing with a few component previews on Histoire. For instance, `Button`, `Expand`, `Item`, `Confirm Modal` & `Window`.

  Also, for the `Checkbox` component, the icon in the checked state wasn't getting rendered correctly.

- A few warnings and an error were getting logged while launching Histoire.
  
  image

Since the PR is blocking a few other ones, maybe we could revisit the issues separately.

**Comment by anwarulislam** [comment]
@jamesgeorge007 
- Unfortunately couldn't reproduce the iPad duplicate nav issue
- I tried to find the element adding extra 2px in height of navbar but couldn't find that any
- Looking into Histoire version

**Review by JoelJacobStephen — APPROVED** [review]
LGTM from the Admin Dashboard side!!

**Review by AndrewBastin — APPROVED** [review]

**Pull request merged**